internal/appsec/remoteconfig.go

```diff
@@ -263,7 +263,11 @@ func mergeRulesData(u remoteconfig.ProductUpdate) ([]ruleDataEntry, map[string]r
 // mergeRulesDataEntries merges two slices of rules data entries together, removing duplicates and
 // only keeping the longest expiration values for similar entries.
 func mergeRulesDataEntries(entries1, entries2 []rc.ASMDataRuleDataEntry) []rc.ASMDataRuleDataEntry {
-	mergeMap := map[string]int64{}
+	count := len(entries1)
+	if count2 := len(entries2); count2 > count {
```

Comment:
There will still be reallocations in most cases since the higher len here will still be shorter than the resulting slice's len, unless it's all duplicates (which is a very unlikely case).
Maybe just ```count := len(entries1) + len(entries2)```?. Uses a bit more memory in case of duplicates but saves on expensive reallocs

Reply:
Oh yes you're right!